src/views/AgmExtension.vue

```diff
@@ -431,12 +428,15 @@ export default class AgmExtension extends Mixins(CommonMixin, DateMixin,
         // convert local properties into API/schema properties
         year: this.data.agmYear,
         isFirstAgm: this.data.isFirstAgm,
-        prevAgmRefDate: this.data.prevAgmDate,
         extReqForAgmYear: this.data.isPrevExtension,
-        expireDateCurrExt: this.data.prevExpiryDate,
         intendedAgmDate: this.data.intendedAgmDate,
         totalApprovedExt: this.data.extensionDuration,
         expireDateApprovedExt: this.data.agmDueDate,
+        // conditionally add properties if not null
+        ...(this.data.prevAgmDate && { prevAgmRefDate: this.data.prevAgmDate }),
+        ...(this.data.prevExpiryDate && {
+          expireDateCurrExt: this.data.prevExpiryDate
+        }),
```

Comment:
The schema validation fails when `prevAgmRefDate = null` or `expireDateCurrExt = null`. It expects it to either be a string or None. To resolve this we need to confitionally add the properties if their respective `data` values exist. 

Syntax is a bit weird, I took inspiration [from here](https://stackoverflow.com/questions/40652364/add-property-to-object-when-its-not-null), but I can move it out of the object assignment into explicit if statements if that'll look cleaner.

Reply:
That's one way to do it.

Another way is to add in the properties outside this block, ie, after this block do "if ... then data.prevAgmDate = xxx".

Another way to do it is to unconditionally add the properties here and then delete them later, ie, after this block do "if (!data.prevAgmDate) delete data.prevAgmDate".

I'll let you pick whichever one you prefer (including the existing code).